spec/consensus/light-client/detection.md

```diff
@@ -1,3 +1,459 @@
+It is not in the interest of faulty full nodes to talk to the failure
+detector as long as the failure detector is connected to at least one
+correct full node. This would only increase the likelihood of
+misbehavior being detected. Also we cannot punish them easily
+(cheaply). The absence of a response need not be the fault of the full
+node. 
+
+Correct full nodes have the incentive to respond, because the failure
+detector may help them to understand whether their header is a good
+one. We can thus base liveness arguments of the failure detector on
+the assumptions that correct full nodes reliably talk to the failure
+detector.
+
+
+**Assumptions**
+
+#### **[LCD-A-CorrFull]**
+
+At all times there is at least one correct full
+node among the primary and the secondary.
+
+*Remark:* Perhaps [LCD-A-CorrFull] is not needed in the end because
+the verification conditions [LCD-VC-*] have preconditions on specific
+cases where primary and/or secondaries are faulty.
+
+#### **[LCD-A-RelComm]**
+
+Communication between the failure detector and a correct full node is reliable and bounded in time.
+
+
+## Problem statement
+
+The failure detector gets as input a header at height *h* and
+queries the secondaries for their headers. Eventually, the failure
+detector should
+decide
+  - whether to report evidence for height *h*
+  - whether to stop operation at height *h*
+
+The failure detector should satisfy the following temporal formulas  
+
+#### **[LCD-VC-INV]**
+
+If there is no fork at height *h*, and the primary
+and the secondaries are correct, then the failure detector should
+never output evidence for height *h* and should not stop at height *h*.
+
+
+#### **[LCD-VC-INV-DONT-STOP]**
+
+If there is no fork at height *h*, and
+ the primary is correct, then the failure detector should never stop
+ at height *h*.
+
+
+#### **[LCD-VC-LIFE-DONT-STOP]**
+
+If there is no fork at height *h*, and
+ the primary is correct, then the failure detector should eventually
+ decide to not stop and to not report evidence
+ at height *h*.
+
+
+#### **[LCD-VC-LIFE-FORK]**
+
+If there is a fork (two correct full nodes decided on different blocks for the same height), and
+- the light client needs to obtain a header of a height *h* that is
+ affected,
+ and
+- there are two correct full nodes *i* and *j* that are
+    - on different branches, and
+    - primary or secondary,
+
+then the failure detector eventually outputs evidence for height *h*.
+
+
+
+#### **[LCD-REQ-REP]**
+
+If the failure detector observes two conflicting headers for height *h*, it should try to verify both. If both are verified it should report evidence.
+
+If the primary reports header *h* and a secondary reports header *h'*,
+     and if *h'* can be verified based on common root of trust, then
+     evidence should be generated; TODO: shall we add "otherwise drop
+     h' and continue normal operation"?
+
+
+## Definitions
+
+- A fixed set of full nodes is provided in the configuration upon
```

Comment:
"Configuration" == "address book" (or includes the address book?) ? Perhaps mention that for consistency sake.